File: php/leet_code/1249.php
<?php

    class Solution
    {

        /**
         * @param String $s
         * @return String
         */
        function minRemoveToMakeValid($s)
        {
            $stack = [];
            $s = str_split($s);
            foreach ($s as $i => $iValue) {
                if ($iValue === '(') {
                    $stack[] = $i;
                } else if ($iValue === ')') {
                    if (count($stack) > 0) {
                        array_pop($stack);
                    } else {
                        $s[$i] = '';
                    }
                }
            }
            foreach ($stack as $i) {
                $s[$i] = '';
            }
            return implode('', $s);
        }
    }

File: swift/leet_code/1249.swift
class Solution {
    func minRemoveToMakeValid(_ s: String) -> String {
        var stack = [Int]()
        var s = Array(s)
        for i in 0 ..< s.count {
            if s[i] == "(" {
                stack.append(i)
            } else if s[i] == ")" {
                if stack.isEmpty {
                    s[i] = " "
                } else {
                    stack.removeLast()
                }
            }
        }
        for i in stack {
            s[i] = " "
        }
        return String(s.filter { $0 != " " })
    }
}
